weight-meter/weight-meter-procedural/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+#![allow(unused_imports)]
+
+use proc_macro::TokenStream;
+use quote::{format_ident, quote, ToTokens};
+use syn::{Attribute, FnArg, Ident, ImplItem, ImplItemMethod, Item, ItemFn, ItemImpl, ItemMod, Pat};
+
+#[proc_macro_attribute]
+pub fn start_with(attr: TokenStream, item: TokenStream) -> TokenStream {
+	let base_weight: syn::Expr = syn::parse(attr).unwrap();
+	let ItemFn { attrs, vis, sig, block } = syn::parse(item).unwrap();
+	(quote! {
+		#(#attrs)*
+		#vis #sig {
+			orml_weight_meter::start_with(#base_weight);
+			let result = #block;
+			orml_weight_meter::finish();
+			result
+		}
+	})
+	.into()
+}
+
+#[proc_macro_attribute]
+pub fn weight(attr: TokenStream, item: TokenStream) -> TokenStream {
+	let weight: syn::Expr = syn::parse(attr).unwrap();
+	let ItemFn { attrs, vis, sig, block } = syn::parse(item).unwrap();
+	(quote! {
+		#(#attrs)*
+		#vis #sig {
+			orml_weight_meter::using(#weight);
+			#block
+		}
+	})
+	.into()
+}
+
+#[proc_macro_attribute]
+pub fn method_benchmarks(_attr: TokenStream, input: TokenStream) -> TokenStream {
+	#[cfg(feature = "runtime-benchmarks")] // Inject methods if we're benchmarking
```

Comment:
could just put all the code with runtime-benchmarks into a mod so we only need one feature gate